src/Symfony/Component/Security/Http/Attribute/IsCsrfTokenValid.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Attribute;
+
+#[\Attribute(\Attribute::IS_REPEATABLE | \Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD | \Attribute::TARGET_FUNCTION)]
```

Comment:
I am not sure what's the difference between METHOD and FUNCTION, but it looks like we only have tests for methods, do we? 🤔

Reply:
@OskarStark I inspired from the IsGranted attribute that apparently supports FUNCTION in addition to METHOD. And FUNCTION wasn't tested for IsGranted case.
Should I add test for it? or better remove the support of FUNCTION for now?

Reply:
If we have a use case add a test case, we should ofc provide a test case for the `IsGranted` in an extra PR too.

Reply:
I do agree there should be a test ideally, at least to avoid regressions especially. But, I'd be fine moving forward without one: here the use case for `TARGET_FUNCTION`  is that controllers can be any callable. Since controllers as callables are test-covered and there is no target-specific behavior with this attribute. We have plently of attributes in the same case including validator constraints.

Reply:
I am fine then 👍

Reply:
I agree, imho we don't necessarily need to test this. This is by-design for all controller argument listeners and the controller argument event has tests to check if we support callables.

Reply:
then this is ready for final review 🙂